src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEth.tsx

```diff
@@ -84,26 +82,52 @@ const VoteReferendum = ({ className, referendumId, onAccountChange, lastVote, se
 		setAvailableWallets(injectedWindow.injectedWeb3);
 		setIsMetamaskWallet((injectedWindow as any)?.ethereum?.isMetaMask);
 	};
+
+	const handleBalanceErr = useCallback(() => {
+		switch (vote){
+		case EVoteDecisionType.AYE:
+			lockedBalance && setIsBalanceErr(availableBalance?.lte(lockedBalance));
+			break;
+		case EVoteDecisionType.NAY:
+			lockedBalance && setIsBalanceErr(availableBalance?.lte(lockedBalance));
+			break;
+		}
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	},[availableBalance, lockedBalance]);
+
 	const handleDefaultWallet=async(wallet:Wallet) => {
 		setWallet(wallet);
-		await getAccounts(wallet);
+		setLoadingStatus({ isLoading: true, message: 'Awaiting accounts' });
+		const accountsData = await getMetamaskAccounts({ chosenWallet: wallet, loginAddress, network });
+		if(accountsData){
+			setAccounts(accountsData?.accounts || []);
+			onAccountChange(accountsData?.account || '');
+			setAddress(accountsData.account);
+			setIsTalismanEthereum(accountsData?.isTalismanEthereum);
+			setLoadingStatus({ isLoading: false, message: 'Awaiting accounts' });
+		}
 		if (walletConnectProvider) {
 			await getWalletConnectAccounts();
 		}
 	};
 
+	const handleModalReset = () => {
+		setLockedBalance(ZERO_BN);
+		ayeNayForm.setFieldValue('balance', ZERO_BN);
+		setLoadingStatus({ isLoading: false, message: '' });
+	};
+
+	useEffect(() => {
+		handleBalanceErr();
+	},[handleBalanceErr]);
+
 	useEffect(() => {
-		if (userDetails.loginWallet && [Wallet.TALISMAN, Wallet.METAMASK].includes(userDetails.loginWallet)) {
-			setLoginWallet(userDetails.loginWallet);
-			setWallet(userDetails.loginWallet);
-		} else {
+		getWallet();
```

Comment:
please confirm if this getWallet with no params or assignment is correct

Reply:
yes